ms2deepscore/models/EmbeddingEvaluatorModel.py

```diff
@@ -0,0 +1,285 @@
+        super().__init__()
+        # Adjust kernel sizes to ensure they are odd for symmetric padding
+        kernel_sizes = [max(kernel_size // (2 ** i), 3) for i in range(3)]
+        kernel_sizes = [k - (k % 2 == 0) for k in kernel_sizes]
+
+        # Use bottleneck layer only if input_channels > 1
+        use_bottleneck = use_bottleneck if input_channels > 1 else False
+        self.bottleneck = nn.Conv1d(input_channels, num_filters, 1, bias=False) if use_bottleneck else nn.Identity()
+        
+        # Prepare convolutional layers with adjusted kernel sizes
+        conv_input_channels = num_filters if use_bottleneck else input_channels
+        self.convs = nn.ModuleList([nn.Conv1d(conv_input_channels, num_filters, k, 
+                                              padding="same") for k in kernel_sizes])
+        
+        # MaxPooling followed by a convolution
+        self.maxconvpool = nn.Sequential(
+            nn.MaxPool1d(3, stride=1, padding=1),
+            nn.Conv1d(input_channels, num_filters, 1, bias=False)
+        )
+        
+        # Batch normalization and activation function
+        self.bn = nn.BatchNorm1d(num_filters * 4)
+
+    def forward(self, x):
+        bottleneck_output = self.bottleneck(x)
+        conv_outputs = [conv(bottleneck_output) for conv in self.convs]
+        pooled_output = self.maxconvpool(x)
+        concatenated_output = torch.cat(conv_outputs + [pooled_output], dim=1)
+        return F.relu(self.bn(concatenated_output))
+
+
+class InceptionBlock(Module):
+    def __init__(self, input_channels, num_filters=32, use_residual=True, depth=6, **kwargs):
+        super().__init__()
+        self.use_residual = use_residual
+        self.depth = depth
+        self.inception_modules = nn.ModuleList()
+        self.shortcuts = nn.ModuleList()
+
+        for d in range(depth):
+            module_input_channels = input_channels if d == 0 else num_filters * 4
+            self.inception_modules.append(InceptionModule(module_input_channels, num_filters, **kwargs))
+            
+            if use_residual and d % 3 == 2:
+                shortcut_input_channels = input_channels if d == 2 else num_filters * 4
+                if shortcut_input_channels == num_filters * 4:
+                    shortcut = nn.BatchNorm1d(shortcut_input_channels) 
+                else:
+                    shortcut = nn.Conv1d(shortcut_input_channels, num_filters * 4, 1, padding="same", bias=False)
+                self.shortcuts.append(shortcut)
+
+    def forward(self, x):
+        residual = x
+        for d in range(self.depth):
+            x = self.inception_modules[d](x)
+            if self.use_residual and d % 3 == 2:
+                shortcut_output = self.shortcuts[d // 3](residual)
+                x = x + shortcut_output
+                x = F.relu(x)
+                residual = x
+        return x
+
+
+class LinearModel:
+    def __init__(self, degree = 2):
```

Comment:
It is not completely clear to me how it is trained. I would expect that compute_embedding_evaluations is used for calculating X. Which results in two predictions for MSE. And Y would be the real MSE. But I am not sure which function is used for that, maybe this can be added to the docstring. Or made part of this class as methods.

Reply:
The linear model will only take pairs of embedding evaluations as input. I added this to the docstring. (we can still make this a bit clearer though when cleaning/merging/wrapping the 3 different models.